backend-server/src/api/controllers/availabilityController.ts

```diff
@@ -136,21 +130,31 @@ export async function makeMultipleCalendarAvailabilities(
     const currStartDate = new Date(availabilities[i].startTime);
     let currEndDate = add(currStartDate, { minutes: availabilityBlockLength });
 
-    for (let j = i + 1; j < availabilities.length; j++) {
-      const nextStartDate = new Date(availabilities[j].startTime);
+    const currIsBooked = availabilities[i].isBooked;
+    const bookedByEmail = currIsBooked ? availabilities[i].bookedByEmail : "";
 
-      if (!isSameMinute(currEndDate, nextStartDate)) {
-        break;
-      }
+    if (!currIsBooked) {
+      for (let j = i + 1; j < availabilities.length; j++) {
+        const nextStartDate = new Date(availabilities[j].startTime);
 
-      currEndDate = add(nextStartDate, { minutes: availabilityBlockLength });
-      i = j;
+        if (
+          !isSameMinute(currEndDate, nextStartDate) ||
+          availabilities[j].isBooked
+        ) {
+          break;
+        }
+
+        currEndDate = add(nextStartDate, { minutes: availabilityBlockLength });
+        i = j;
+      }
     }
 
     const newCalendarAvailability: CalendarAvailability = {
       interviewerUID: availabilities[i].interviewerUID,
       start: formatISO(currStartDate),
       end: formatISO(currEndDate),
+      isBooked: currIsBooked,
+      bookedByEmail,
```

Comment:
Should this be `bookedByEmail: bookedByEmail,`?

Reply:
it can, but this is one of those [JS/TS shorthands](https://alligator.io/js/object-property-shorthand-es6/) -- if an object key name whose value is held in a variable of the same name (which is the case with the key `bookedByEmail` and the variable `bookedByEmail`), you can write that entire key, value pair as just the variable name and it gets understood automagically 🪄